types/serialization.go

```diff
@@ -0,0 +1,37 @@
+package types
+
+import (
+	pb "github.com/lazyledger/optimint/types/pb/optimint"
+)
+
+type Serializable interface {
+	Serialize() ([]byte, error)
+}
```

Comment:
This seems very odd to me, as Golang already has interfaces and primitives for marshaling. Pls use encoding.BinaryMarshaler instead.

Reply:
Good point! I was pretty sure that there should be something "already", but couldn't find it.